src/App.tsx

```diff
@@ -1,24 +1,240 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect, useMemo, useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { TodoList } from './components/TodoList';
+import { Form } from './components/Form';
+import { Footer } from './components/Footer';
+import { Notifications } from './components/Notifications';
+import { Todo } from './types/Todo';
+import * as todoService from './api/todos';
+import * as TodoFilter from './utils/TodoFilter';
+import { FilterType } from './types/FilterType';
+import { NOTIFICATION } from './types/Notification';
+import { USER_ID } from './constants/USER_ID';
+import { findTodoById } from './utils/FindPostById';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.ALL);
+  const [notification, setNotification] = useState(NOTIFICATION.CLEAR);
+  const [loading, setLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [listOfTodoId, setListOfTodoId] = useState<number[]>([]);
+  const [updatingTodos, setUpdatingTodos] = useState<number[]>([]);
+  const [titleFocus, setTitleFocus] = useState(true);
+
+  useEffect(() => {
+    todoService.getTodos(USER_ID)
+      .then((data) => {
+        setTodos(data);
+        setTempTodo(null);
+      }).catch(() => setNotification(NOTIFICATION.LOAD));
+  }, []);
+
+  useEffect(() => {
+    setListOfTodoId([]);
+    setUpdatingTodos([]);
+  }, [todos]);
```

Comment:
why are you setting todos to empty lists when todos are changing?

Reply:
> why are you setting todos to empty lists when todos are changing?

fixed